flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/batch/sql/CorrelateTest.scala

```diff
@@ -43,7 +43,7 @@ class CorrelateTest extends TableTestBase {
         term("invocation", "func1($cor0.c)"),
         term("function", func1.getClass.getCanonicalName),
         term("rowType",
-             "RecordType(INTEGER a, BIGINT b, VARCHAR(2147483647) c, VARCHAR(2147483647) f0)"),
+             "RecordType(INTEGER a, BIGINT b, VARCHAR(65536) c, VARCHAR(65536) f0)"),
```

Comment:
It seems that Calcite has defined an upper limit for strings and ignores our size. Can you also update this in `FlinkTypeSystem` and the comment in `FlinkTypeFactory#createSqlType`?